types/dependency-cruiser.d.ts

```diff
@@ -406,13 +406,451 @@ export interface ICruiseOptions {
+   * core
+   */
+  coreModule: boolean;
+  /**
+   * 'true' if dependency-cruiser could not resulve the module name in the source code to a
+   * file name or core module. 'false' in all other cases.
+   */
+  couldNotResolve: boolean;
+  /**
+   * If following this dependency will ultimately return to the source (circular === true),
+   * this attribute will contain an (ordered) array of module names that shows (one of the)
+   * circular path(s)
+   */
+  cycle?: string[];
+  /**
+   * the type of inclusion - local, core, unknown (= we honestly don't know), undetermined (=
+   * we didn't bother determining it) or one of the npm dependencies defined in a package.jsom
+   * ('npm' for 'depenencies', 'npm-dev', 'npm-optional', 'npm-peer', 'npm-no-pkg' for
+   * development, optional, peer dependencies and dependencies in node_modules but not in
+   * package.json respectively)
+   */
+  dependencyTypes: DependencyType[];
+  /**
+   * true if this dependency is dynamic, false in all other cases
+   */
+  dynamic: boolean;
+  /**
+   * Whether or not this is a dependency that can be followed any further. This will be
+   * 'false' for for core modules, json, modules that could not be resolved to a file and
+   * modules that weren't followed because it matches the doNotFollow expression.
+   */
+  followable: boolean;
+  /**
+   * the license, if known (usually known for modules pulled from npm, not for local ones)
+   */
+  license?: string;
+  /**
+   * 'true' if the file name of this module matches the doNotFollow regular expression
+   */
+  matchesDoNotFollow?: boolean;
+  /**
+   * The name of the module as it appeared in the source code, e.g. './main'
+   */
+  module: string;
+  moduleSystem: ModuleSystemType;
+  /**
+   * The (resolved) file name of the module, e.g. 'src/main//index.js'
+   */
+  resolved: string;
+  /**
+   * an array of rules violated by this dependency - left out if the dependency is valid
+   */
+  rules?: IRuleSummaryType[];
+  /**
+   * 'true' if this dependency violated a rule; 'false' in all other cases. The violated rule
+   * will be in the 'rule' object at the same level.
+   */
+  valid: boolean;
+}
+
+/**
+ * If there was a rule violation (valid === false), this object contains the name of the
+ * rule and severity of violating it.
+ */
+export interface IRuleSummaryType {
+  /**
+   * The (short, eslint style) name of the violated rule. Typically something like
+   * 'no-core-punycode' or 'no-outside-deps'.
+   */
+  name: string;
+  severity: SeverityType;
+}
+
+/**
+ * How severe a violation of a rule is. The 'error' severity will make some reporters return
+ * a non-zero exit code, so if you want e.g. a build to stop when there's a rule violated:
+ * use that. The absence of the 'ignore' severity here is by design; ignored rules don't
+ * show up in the output.
+ *
+ * Severity to use when a dependency is not in the 'allowed' set of rules. Defaults to 'warn'
+ */

```

Comment:
this piece of jsdoc pertains to the `severity` in `IRuleSummaryType` - can you move it up there?

Reply:
Sorry, Cant able to get this comment!

Reply:
I think the jsdoc belongs just over the `severity` attribute, like so:

```javascript
export interface IRuleSummaryType {
   /**
    * The (short, eslint style) name of the violated rule. Typically something like
    * 'no-core-punycode' or 'no-outside-deps'.
    */
   name: string;
   /**
    * How severe a violation of a rule is. The 'error' severity will make some reporters return
    * a non-zero exit code, so if you want e.g. a build to stop when there's a rule violated:
    * use that. The absence of the 'ignore' severity here is by design; ignored rules don't
    * show up in the output.
    *
    * Severity to use when a dependency is not in the 'allowed' set of rules. Defaults to 'warn'
    */
   severity: SeverityType;
 }
```

Reply:
sure got it